### PR TITLE
Switch sensor/schedule view to show tick list instead of run list

### PR DESCRIPTION
**Pull request by prha**
## Summary

There aren't good tools to debug / investigate schedule/sensor ticks.  There also is not a good way to dig into cursor values associated with sensors.  This PR adds a flattened list view of ticks to the schedule / sensor page, allowing users to page through history, where previously the only view available was the timeline view.

For real estate, we removed the sensor runs / schedule runs table, but you can get a view of the runs by filtering the tick list to the "Success" status ticks.

https://user-images.githubusercontent.com/1040172/156270257-848b558f-8d23-4db0-9990-460a054b4955.mov

### CODE REVIEW

**Comment by sryza** [comment]
This change makes sense to me. A couple thoughts:
* Maybe we should default to not checking the skipped checkbox? I'm worried about this page not being useful at first glance for most sensors.
* If we include the checkbox state in the URL, it would enable users to bookmark a page that includes just the Started ticks, if that's what they care about.
* It could make sense to provide a link to a Runs view that's filtered down to runs that were triggered by the schedule.

**Comment by prha** [comment]
Yep, hiding skips, persisting flags in the URL and including seconds in the timestamp all sound reasonable and easy.

Main challenge with adding a Run link for me is figuring out where to put it.  Same with an action menu that allows the user to set the cursor.  (cc @braunjj)

Re: toggling timeline / list view, I think that makes sense, but it throws the balance of the page off.   I couldn't get it to feel right.

**Comment by braunjj** [comment]
Love this change! I mocked up the refinement ideas we discussed this morning below:
[View the full Figma file here](https://www.figma.com/file/ITO0KOQ7ljh9Dtru5o603o/Sensor-Page-Update?node-id=0%3A1)

**Tick History Tab (Default)**
image
- Add Edit Cursor button at the top
- Remove the timeline view filters and only show "Recent Ticks" in the timeline
- Add a "View Errors" button next to error status

**Run History Tab **
image
- Users can tab over to the run history tab to view all recent runs from this sensor

**Edit Cursor Dialog**
image

**Comment by prha** [comment]
instead of closing, created some base branches and restacked.

**Comment by prha** [comment]
Latest update:

https://user-images.githubusercontent.com/1040172/156677357-16e176ff-8049-430a-bb4c-990d45343744.mov

File: js_modules/dagit/packages/core/src/schedules/ScheduleRoot.tsx [hunk 1]
```diff
@@ -123,10 +124,16 @@ const SchedulePreviousRuns: React.FC<SchedulePreviousRunsProps> = (props) => {
     pollInterval: 15 * 1000,
   });
 
+  if (!data || data.pipelineRunsOrError.__typename !== 'Runs') {
+    return null;
+  }
```
Comment (gibsondan) [inline js_modules/dagit/packages/core/src/schedules/ScheduleRoot.tsx:129]: are we swallowing here if there's an error? should we display it somewhere?

File: js_modules/dagit/packages/core/src/sensors/EditCursorDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  const onSave = async () => {
+    setIsSaving(true);
+    const {data} = await requestSet({
+      variables: {sensorSelector, cursor: cursorValue},
+    });
+    if (data?.setSensorCursor.__typename === 'Sensor') {
+      SharedToaster.show({message: 'Set cursor value', intent: 'success'});
+    } else {
+      SharedToaster.show({message: 'Could not set cursor value', intent: 'danger'});
+    }
```
Comment (gibsondan) [inline js_modules/dagit/packages/core/src/sensors/EditCursorDialog.tsx:29]: similar here - are we swallowing, should we display? just thinking about debugging if this fails

I guess the new capture_errors behavior in python would show the error in the dagit logs maybe?

File: js_modules/dagit/packages/core/src/sensors/SensorPreviousRuns.tsx [hunk 3]
```diff
@@ -27,10 +28,16 @@ export const SensorPreviousRuns: React.FC<{
     },
   });
 
+  if (!data || data.pipelineRunsOrError.__typename !== 'Runs') {
+    return null;
```
Comment (gibsondan) [inline js_modules/dagit/packages/core/src/sensors/SensorPreviousRuns.tsx:32]: same comment as above

File: python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_sensors.py [hunk 4]
```diff
@@ -267,6 +339,37 @@ def test_stop_sensor(self, graphql_context):
             == InstigatorStatus.STOPPED.value
         )
 
+    def test_set_cursor(self, graphql_context):
```
Comment (gibsondan) [inline python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_sensors.py:342]: for extra credit, add a test case on an automatically on sensor with no rows in the DB yet

**Review by gibsondan — APPROVED** [review]
- its a little surprising to me that the timeline view only shows up when the sensor/schedule is running (i could imagine that viz being useful for a stopped sensor/schedule that previously produced ticks) but I'll defer to @braunjj on that
- copy nit: consider changing the "Set cursor value" confirmation text to something like "Cursor value updated"
- just confirming that the changes to schedule storage here have tests in TestScheduleStorage that will also run in internal

**Pull request merged**